src/internal/util/NodeEventEmitterDataType.ts

```diff
@@ -0,0 +1,236 @@
+/**
+ * Issue #6669 - fix(fromEvent): infer from Node.js EventEmitter with types
+ *  @see https://github.com/ReactiveX/rxjs/pull/6669
+ *
+ *  Author: Huan <https://github.com/huan>
+ *  Date: Nov 7, 2021
+ *
+ * TypeScript has design limitation that prevents us from inferring from the overload functins:
+ *  > ReturnType or the use of infer in a function parameter or return position does handle overloads.
+ *    the last overload is used for inference as it is assumed to be the most general.
+ *    - @mhegazy @link https://github.com/Microsoft/TypeScript/issues/24275#issuecomment-390701982
+ *
+ * This file is created by @huan and it was inspired from the brilliant work from @Aidin on StackOverflow:
+ *  @link https://stackoverflow.com/a/60822641/1123955
+ */
+
+/* eslint-disable no-use-before-define */
+/* eslint-disable max-len */
+
+/**
+ * Node.js EventEmitter Add/Remove Listener interface
+ */
+interface L<N, D> {
```

Comment:
It would be great if we used more descriptive names here.

Reply:
I agree that the descriptive name will be more readable, and at first, I was using the descriptive name.

However, the reason that I finally use the short name (`L` for Listener, `N` for Name, and `D` for Data) is that the below code will have lots of repeated those type names:

For example:

```ts
type EventNameDataPair9<AddRemoveListener> = AddRemoveListener extends L9<
  infer N1,
  infer N2,
  infer N3,
  infer N4,
  infer N5,
  infer N6,
  infer N7,
  infer N8,
  infer N9,
  infer D1,
  infer D2,
  infer D3,
  infer D4,
  infer D5,
  infer D6,
  infer D7,
  infer D8,
  infer D9
>
  ? [N1, D1] | [N2, D2] | [N3, D3] | [N4, D4] | [N5, D5] | [N6, D6] | [N7, D7] | [N8, D8] | [N9, D9]
  : never;
```

The above code will be very long and I feel the short version of the name is more readable, so I'd like to suggest that we can keep this short name because they are very to be understood:

1. `L` - Listener
2. `N` - Name
3. `D` - Data

On the other hand, I'm OK with using a more descriptive(long) name if you think the long version is better.

Please let me know your final decision and I'll follow it.